### PR TITLE
Synchronize the player's cyclic inventory and capabilities

### PR DESCRIPTION
**Pull request by bluelightning32**

### CODE REVIEW

File: src/main/java/world/jnc/invsync/permission/PermissionRegistry.java [hunk 1]
```diff
@@ -66,6 +66,11 @@ public void registerPermissions() {
           PlayerSerializer.spiceOfLifeSyncModule.getPermission(),
           "Allow this user's Spice of Life food list to be synchronized",
           PermissionDescription.ROLE_USER);
+    if (PlayerSerializer.cyclicSyncModule.canBeEnabled())
```
Comment (BrainStone) [inline src/main/java/world/jnc/invsync/permission/PermissionRegistry.java]: Can you please add it in alphabetical order?

File: src/main/java/world/jnc/invsync/util/serializer/PlayerSerializer.java [hunk 2]
```diff
@@ -69,7 +71,8 @@
               hungerSyncModule,
               potionEffectsSyncModule,
               baublesSyncModule,
-              spiceOfLifeSyncModule));
+              spiceOfLifeSyncModule,
+              cyclicSyncModule));
```
Comment (BrainStone) [inline src/main/java/world/jnc/invsync/util/serializer/PlayerSerializer.java]: Same here.

File: src/main/java/world/jnc/invsync/util/serializer/module/mod/CyclicSyncModule.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        props.setInventoryExtended(serializedData.get().getBoolean(INVENTORY_CAPABILITY).get());
+        CapabilityRegistry.syncServerDataToClient((EntityPlayerMP) nativePlayer);
+
+        // Create a new inventory to deserialize into. This way if anything goes wrong, the current
+        // extended inventory isn't overwritten.
+        InventoryPlayerExtended inventory = new InventoryPlayerExtended(nativePlayer);
+
+        boolean fail =
+            NativeInventorySerializer.deserializeInventory(
+                serializedData.get().getViewList(INVENTORY).get(), new InventoryAdapter(inventory));
```
Comment (BrainStone) [inline src/main/java/world/jnc/invsync/util/serializer/module/mod/CyclicSyncModule.java]: Store the queries in an Optional please and print them during debug. Debug message shlould also be pretty late.

File: src/main/java/world/jnc/invsync/permission/PermissionRegistry.java (hunk 1 above)
Comment (bluelightning32) [reply src/main/java/world/jnc/invsync/permission/PermissionRegistry.java]: The other mods aren't in alphabetical order. So I took a guess at what you meant.

**Pull request merged**